src/libraries/Microsoft.Extensions.Options/gen/Emitter.cs

```diff
@@ -491,25 +491,31 @@ public override bool IsValid(object? value)
         {
             if (!Initialized)
             {
-                if (Minimum is null || Maximum is null)
-                {
-                    throw new global::System.InvalidOperationException(c_minMaxError);
-                }
-                if (NeedToConvertMinMax)
+                lock (this)
                 {
-                    System.Globalization.CultureInfo culture = ParseLimitsInInvariantCulture ? global::System.Globalization.CultureInfo.InvariantCulture : global::System.Globalization.CultureInfo.CurrentCulture;
+                    if (!Initialized)
+                    {
+                        if (Minimum is null || Maximum is null)
+                        {
+                            throw new global::System.InvalidOperationException(c_minMaxError);
+                        }
+                        if (NeedToConvertMinMax)
+                        {
+                            System.Globalization.CultureInfo culture = ParseLimitsInInvariantCulture ? global::System.Globalization.CultureInfo.InvariantCulture : global::System.Globalization.CultureInfo.CurrentCulture;
 {{initializationString}}
+                        }
+                        int cmp = ((global::System.IComparable)Minimum).CompareTo((global::System.IComparable)Maximum);
+                        if (cmp > 0)
+                        {
+                            throw new global::System.InvalidOperationException("The maximum value '{Maximum}' must be greater than or equal to the minimum value '{Minimum}'.");
+                        }
+                        else if (cmp == 0 && (MinimumIsExclusive || MaximumIsExclusive))
+                        {
+                            throw new global::System.InvalidOperationException("Cannot use exclusive bounds when the maximum value is equal to the minimum value.");
+                        }
+                        Initialized = true;
```

Comment:
I don't see anything that would prevent this write from moving earlier, in which case on an architecture like Arm another thread could end up reading Initialized as true before all the checks had been performed.  The fix is to use volatile for the read and write.  That would necessitate making Initialized into a field, which seems fine... as it's private, I'd have expected it to be an _initialized field rather than a property, anyway.

Reply:
I addressed this feedback. Please let me know if you are fine with the change. Thanks!